scripts/publish-packages.sh

```diff
@@ -1,16 +1,37 @@
+#!/bin/sh
+
+set -eu
+
 SCRIPTS_DIR="$(dirname "$0")"
 cd "${SCRIPTS_DIR}/../packages"
 for d in */ ; do
-    cd $d
+    cd "$d"
 
+    # Get the current package name and version from package.json.
     PACKAGE_NAME=$(node -p "require('./package.json').name")
     PACKAGE_VERSION=$(node -p -e "require('./package.json').version")
     echo "Will check if version ${PACKAGE_VERSION} of ${PACKAGE_NAME} should be published."
 
-    PUBLISHED_VERSION=$(npm view "${PACKAGE_NAME}" version)
+    # Look up the currently published version of the package in the registry.
+    # npm view will fail if the package does not exist. This case needs to be handled
+    # in order to support publishing new packages for the first time.
+    set +e
+    PUBLISHED_VERSION=$(npm --loglevel=error view "${PACKAGE_NAME}" version 2>&1)
```

Comment:
`--loglevel=error` ignores warnings about "Ignoring workspaces for specified package(s)".